cmd/sealer/cmd/join.go

```diff
@@ -57,7 +59,27 @@ join default cluster:
 			return err
 		}
 
-		//TODO mount image and copy to new hosts
+		imageEngine, err := imageengine.NewImageEngine(imagecommon.EngineGlobalConfigurations{})
+		if err != nil {
+			return err
+		}
+
+		distributor, err := imagedistributor.NewScpDistributor(imageEngine, infraDriver)
+		if err != nil {
+			return err
+		}
+
+		var (
+			clusterImageName = cluster.Spec.Image
+			hosts            = append(newMasters, newWorkers...)
+		)
+
+		// distribute rootfs
+		if err = distributor.Distribute(clusterImageName, imagedistributor.FilterOptions{
```

Comment:
i think maybe we can rename func Distribute to DistributeRootfsDir.

in cmd, we just run DistributeRootfsDir, which only distribute rootfs except registry dir. And we give infradriver as a parameter of RegistryConfigurator, than RegistryConfigurator will use infradriver to copy registry dir or other things, depends on its own implementation

Reply:
Because "			ExceptDirs: []string{"registry"}," is very confused.